### PR TITLE
Add populate query

### PR DESCRIPTION
**Pull request by garyb**
@jonsterling we need this so we can restore the state of a markdown cell. This doesn't exactly validate that the set values are correct (so we could in theory set an unselectable value in a dropdown), but at least will prune irrelevant entries.

### CODE REVIEW

**Comment by garyb** [comment]
Oh, I added the pursuit-publish-on-release thing too.

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
 script:
   - gulp
+after_success:
+- >-
+  test $TRAVIS_TAG &&
+  node_modules/.bin/psc-publish > .pursuit.json &&
+  curl -X POST http://pursuit.purescript.org/packages \
+    -d @.pursuit.json \
+    -H 'Accept: application/json' \
+    -H "Authorization: token ${GITHUB_TOKEN}"
```
Comment (jonsterling) [inline .travis.yml:18]: nice.

File: src/Text/Markdown/SlamDown/Html.purs [hunk 2]
```diff
@@ -215,6 +217,13 @@ evalSlamDownQuery e =
     GetFormState k -> do
       SlamDownState state <- H.get
       pure $ k state.formState
+    PopulateForm values next -> do
+      SlamDownState { document } <- H.get
+      let desc = formDescFromDocument document
+          keysToPrune = filter (`M.member` desc) (toList (M.keys values))
+          prunedFormState = foldr M.delete values keysToPrune
```
Comment (jonsterling) [inline src/Text/Markdown/SlamDown/Html.purs]: I'm a little confused by the logic here. It looks like it's finding out which keys of `values` are already part of the document, and then removing them from `values` prior to updating the state. Is this what we want to do, or have I got turned around somehow?

Comment (garyb) [inline src/Text/Markdown/SlamDown/Html.purs]: No you're right, I was rushing and the logic here is inverted. Thanks for the spot!

Comment (jonsterling) [inline src/Text/Markdown/SlamDown/Html.purs]: no problem! :smile:

File: src/Text/Markdown/SlamDown/Html.purs [hunk 3]
```diff
@@ -215,6 +217,13 @@ evalSlamDownQuery e =
...
+      H.modify <<< modifySlamDownState $ const prunedFormState
```
Comment (jonsterling) [inline src/Text/Markdown/SlamDown/Html.purs]: Is it possible to replace the `get` followed by `modify` with a single call to `modify` without `const`?

**Comment by garyb** [comment]
All good calls, done now. I've also updated the README so it has the link to the pursuit docs, and latest version and dependencies badges.

File: src/Text/Markdown/SlamDown/Html.purs [hunk 4]
```diff
@@ -215,6 +217,13 @@ evalSlamDownQuery e =
     GetFormState k -> do
       SlamDownState state <- H.get
       pure $ k state.formState
+    PopulateForm values next -> do
+      H.modify \(SlamDownState { document }) ->
+        let desc = formDescFromDocument document
+            keysToPrune = filter (\newKey -> not (newKey `M.member` desc)) (toList (M.keys values))
+            prunedValues = foldr M.delete values keysToPrune
+        in SlamDownState { document, formState: prunedValues }
```
Comment (jonsterling) [inline src/Text/Markdown/SlamDown/Html.purs:225]: this new record punning stuff is pretty slick.

**Comment by jonsterling** [comment]
@garyb thanks! I'm happy with this PR, and will mark it ready once Travis gives the OK. :+1:

**Comment by garyb** [comment]
Cool, thanks! And yeah, record punning is almost indispensable already :smile:

**Pull request merged**